src/fortran_routing/Reservoir_Binding/kernels/fortran/Reservoirs/bind_hybrid.f90

```diff
@@ -0,0 +1,154 @@
+
+        SUBROUTINE init_hybrid(handle,  water_elevation,  &
+        lake_area, weir_elevation, weir_coeffecient, &
+        weir_length, dam_length, orifice_elevation, orifice_coefficient, &
+        orifice_area, lake_max_water_elevation, initial_fractional_depth, lake_number, reservoir_type, reservoir_parameter_file, start_date, &
+        usgs_timeslice_path, usace_timeslice_path, observation_lookback_hours, &
+        observation_update_time_interval_seconds) BIND(C, NAME='init_hybrid')
+            USE, INTRINSIC :: ISO_C_BINDING, ONLY: C_PTR, C_F_POINTER, C_CHAR, c_null_char
+            TYPE(C_PTR), INTENT(IN), VALUE :: handle
+            real,    intent(inout) :: water_elevation           ! meters AMSL
+            real,    intent(in)    :: lake_area      		    ! area of lake (km^2)
+            real,    intent(in)    :: weir_elevation            ! bottom of weir elevation (meters AMSL)
+            real,    intent(in)    :: weir_coeffecient          ! weir coefficient
+            real,    intent(in)    :: weir_length               ! weir length (meters)
+            real,    intent(in)    :: dam_length                ! dam length (meters)
+            real,    intent(in)    :: orifice_elevation         ! orifice elevation (meters AMSL)
+            real,    intent(in)    :: orifice_coefficient       ! orifice coefficient
+            real,    intent(in)    :: orifice_area              ! orifice area (meters^2)
+            real,    intent(in)    :: lake_max_water_elevation  ! max water elevation (meters)
+            real,    intent(in)    :: initial_fractional_depth  ! initial fraction water depth
+            integer, intent(in)    :: lake_number               ! lake number
+            integer, intent(in)    :: reservoir_type            ! reservoir type
+            character(kind=c_char), dimension(*), intent(IN) :: reservoir_parameter_file
+            character(kind=c_char), dimension(*), intent(IN) :: start_date
+            character(kind=c_char), dimension(*), intent(IN) :: usgs_timeslice_path
+            character(kind=c_char), dimension(*), intent(IN) :: usace_timeslice_path
+            integer,            intent(in) :: observation_lookback_hours
+            integer,            intent(in) :: observation_update_time_interval_seconds
+
+            type (persistence_levelpool_hybrid), POINTER :: hybrid_ptr ! ptr to hybrid object
+
+            character(len=256) :: reservoir_parameter_file_F
+            character(len=256) :: start_date_F
+            character(len=256) :: usgs_timeslice_path_F
+            character(len=256) :: usace_timeslice_path_F
+
+            integer :: char_index, nchars_reservoir_parameter_file, nchars_start_date, nchars_usgs_timeslice_path, nchars_usace_timeslice_path
+         
+            ! Use fortran calls to go find certain things like null_char instead of using this loop every time
+            ! Look up more about converting char array to string
+
+            do char_index = 1, 256 
```

Comment:
I spent some more time looking for alternatives to these loops. I came up with no alternatives yet but will return to these later.